bolt-taproot-gossip.md

```diff
@@ -0,0 +1,1083 @@
+    `announcement_sig` field (see the
+    [`MsgHash`](#signature-message-construction) definition).
+- otherwise:
+    - MUST NOT send the `announcement_signatures_2` message.
+- upon reconnection (once the above timing requirements have been met):
+    - MUST respond to the first `announcement_signatures_2` message with its own
+      `announcement_signatures_2` message.
+    - if it has NOT received an `announcement_signatures_2` message:
+        - SHOULD retransmit the `announcement_signatures_2` message.
+
+A recipient node:
+- if the `short_channel_id` is NOT correct:
+    - SHOULD send a `warning` and close the connection, or send an
+      `error` and fail the channel.
+- if the `partial_signature` is NOT valid as
+  per [Partial Signature Verification](#partial-signature-verification):
+    - MAY send a `warning` and close the connection, or send an
+      `error` and fail the channel.
+- if it has sent AND received a valid `announcement_signatures_2` message:
+    - SHOULD queue the `channel_announcement_2` message for its peers.
+- if it has not sent `channel_ready`:
+    - MAY send a `warning` and close the connection, or send an `error` and fail
+      the channel.
+
+### Rationale
+
+The message contains the necessary partial signature, by the sender, that the
+recipient will be able to combine with their own partial signature to construct
+the signature to put in the `channel_announcement_v2` message. Unlike the legacy
+`announcement_signatures` message, `announcement_signatures_v2` only has one
+signature field. This field is a MuSig2 partial signature which is the
+aggregation of the two signatures that the sender would have created (one for
+`bitcoin_key_x` and another for `node_ID_x`).
+
+## The `channel_announcement_2` Message
+
+This gossip message contains ownership information regarding a taproot channel. 
+It ties each on-chain Bitcoin key that makes up the taproot output key to the 
+associated Lightning node key, and vice-versa. The channel is not practically 
+usable until at least one side has announced its fee levels and expiry, using 
+`channel_update_2`.
+
+See [Taproot Channel Proof and Verification](#taproot-channel-proof-and-verification)
+for more information regarding the requirements for proving the existence of a
+channel.
+
+1. type: xxx (`channel_announcement_2`)
+2. data: (tlv_stream):
+   1. type: 0 (`announcement_sig`)
+   2. data:
+        * [`bip340_sig`:`bip340_sig`]
+   3. type: 1 (`features`)
+   4. data:
+        * [`...*byte`: `features`]
+   5. type: 2 (`chain_hash`)
+   6. data:
+        * [`chain_hash`:`chain_hash`]
+   7. type: 3 (`short_channel_id`)
+   8. data:
+        * [`short_channel_id`:`short_channel_id`]
+   9. type: 4 (`node_id_1`)
+   10. data:
+        * [`point`:`point`]
+   11. type: 5 (`node_id_2`)
+   12. data:
+       * [`point`:`point`]
+   13. type: 6 (`bitcoin_key_1`)
+   14. data:
+       * [`point`:`point`]
+   15. type: 7 (`bitcoin_key_2`)
```

Comment:
See my response to Rusty's comment above. Basically: I think we can go full-ghost mode once we do Gossip 2.0 (overcommit stuff) but before that (gossip 1.5 aka this) we should have at least _some_ binding to the LN context. The way the proposal is currently designed, nodes can prove that the output cant be spent along the script path. Yes, perhaps we can smush the 2 bitcoin keys into one (i would just need to check that the musig verification can still work)

Reply:
> Basically: I think we can go full-ghost mode once we do Gossip 2.0 (overcommit stuff) but before that (gossip 1.5 aka this) we should have at least some binding to the LN context.  The way the proposal is currently designed, nodes can prove that the output cant be spent along the script path.

IIUC, it's important that node1 and node2 prove to *each other* that script path spends are impossible.  But I don't see a reason we need to prove that to the rest of the network.

Reply:
> it's important that node1 and node2 prove to each other that script path spends are impossible. But I don't see a reason we need to prove that to the rest of the network.

i don't think they need to prove this to eachother. They would not have generated the same output key & hence would not have produced the correct signatures if either node added a script path. They would not be able to hide such a path from the peer. 

The reason for the "proving to the network" part is similar to today's reason for advertising the 2 keys: today we prove that the output can only be spent via a 2-of-2 multisig? this makes it a _bit_ more difficult for an attacker to create fake spam since they are now limited in how they can spend the utxo since they would now have to pay more for spending from it (2 sigs). That disincentive is not available with taproot - so the idea here is that we at least show the network that we are limiting how the utxo can be spent in _some_ way - by showing that we cant spend via a script path. But perhaps this is too "cosmetic" to consider sufficient in any case. 

So i think the main thing we should iron out is: do we want to be able to prove (ie, do we care) that the utxo is created via a 2-of-2 musig2 or not. Perhaps a good topic for today's spec call

Reply:
> That disincentive is not available with taproot - so the idea here is that we at least show the network that we are limiting how the utxo can be spent in some way - by showing that we cant spend via a script path. But perhaps this is too "cosmetic" to consider sufficient in any case.

The "no-script-path" limitation seems insignificant to me.  I suspect most P2TR outputs today don't have a script path anyway.

> So i think the main thing we should iron out is: do we want to be able to prove (ie, do we care) that the utxo is created via a 2-of-2 musig2 or not.

In the taproot world there's no fee difference between musig and single sig UTXOs.  So spam prevention doesn't seem a compelling reason to prove a UTXO is a musig.

It seems to me that proof-of-UTXO is a "good enough" spam prevention.  In fact, we're talking about moving to overcommit in the future, which would be a weaker spam prevention than is being proposed right here.

Reply:
yep - so I think the next version is gonna make this tweak optional. So by default, nodes will just use the output key found on chain & then if nodes want to include things like the 2 keys used in the musig2 & if they want to add a tweak then they can do so optionally.